README.md
[![Travis-CI Build Status](https://travis-ci.org/mexindian/TileMaker.svg?branch=master)](https://travis-ci.org/mexindian/TileMaker)

# TileMaker
An R script that enables the creation of data tiles for inclusion in a html dashboard or some such.

## Now available as a package!
To install, please type:
```
devtools::install_github("mexindian/TileMaker")
```



## Example:

```
library(TileMaker)
Button1 <- ButtonMaker(Color = 2,Value = 3.57,Subtitle = "Times apple eaten")
Button2 <- ButtonMaker(Color = 3,Value = 13.7,Subtitle = "Nutritional value")
Button3 <- ButtonMaker(Color = 4,Value = 1,Subtitle = "Yumminess factor")
Button4 <- ButtonMaker(Color = 5,Size=1,Value = 5,Subtitle = "Inconsistencies")

Div1 <- DivMaker(Title = "Quantativity factors",Buttons = paste(Button1,Button2))
Div2 <- DivMaker(Title = "Implementation procedures",Buttons = paste(Button3,Button4))
TileMaker(MainTitle = "Hello",Divs = paste(Div1,Div2),FileName = "a.html")
browseURL("a.html")
```
This will give something like this (every element of this is customizable):

![Example](https://github.com/mexindian/TileMaker/blob/master/example.PNG)

For latest live version, please see here: http://htmlpreview.github.io/?https://github.com/mexindian/TileMaker/blob/master/example.html

## Arguments
### TileMaker 
**MainTitle**        Optional title for the whole set of titles

**Divs**              The Divs that you want inserted into the tile. If you have more than one div, use paste(Div1,Div2)

**FileName**          The filename the tile should spit out as, including the extension

**ShowDate**          Optional boolean controlling whether the date should be included or suppressed

**localCSS**          Optional boolean to specify whether the bootstrap css file should be served from the internet, or if you have saved                   a local version. If you have saved a local version, make sure to download the "fonts" folder too, otherwise                           glyphicons won't work.


### DivMaker 
**Title**             The title for this row of buttons

**Buttons**           The Buttons that you want inserted into this row. If you have more than one button, use paste(Button1,Button2)


### ButtonMaker
**Color**             Optional numeric 1-6, corresponding to the colors specified in the bootstrap css classes: 
                  "success",  "warning", "danger", "info", "primary", "default"

**Size**              Optional numeric 1-4, corresponding to the sizes specified in the bootstrap css classes:
                  "xs","sm","md","lg")

**Value**             The numeric value you want to highlight (the main enchilada)

**Subtitle**          Optional subtext that should appear under the value

**Link**             Optional hyperlink that should be followed on click

**Icon**              Optional glyphicon that should be displayed from http://getbootstrap.com/components/

**Units**             Optional units that should be displayed after Value

**Target**            Optional target that the value should be compared against. Use with ThresholdHigh and THresholdLow

**ThresholdHigh**     Optional border between "green" and "orange". Use w/ Target and ThresholdLow. This value represents the RATIO
                  of the VALUE to the TARGET that, if above the ThresholdHigh will show as green, and if not, as orange

**ThresholdLow**      Optional border between "orange" and "red". Use w/ Target and ThresholdLow. This value represents the RATIO
                  of the VALUE to the TARGET that, if above the ThresholdHigh will show as orange, and if not, as red
                        
**Hover**			  Optional tooltip, or text that will show up when a user rests their mouse over the button.                      

**alpha**             Optional transparency coefficient for the icon, a decimal from 0-1.


NEWS
0.1.0
==============

Converting function to stand-alone package


0.2.0
==============
 - Created all new functions using htmlTools. All old function names are depreciated
 - Seperated simple box from one that responds to threshold to clarify better hat imputs are required to each
 - TileMaker now doesn't need a filename
 - Added tile_matrix function to make multiple buttons suitable to quickly take inputs from a dataframe
 - Added multi_box function to create buttons that contain more than one value and icon.
 - CRAN-ready version


R/tilemaker.R

## Make a html file containing tiles for single data point visualization using bootstrap
## This includes 3 functions that should be used in sequence:
## TileMaker(
##   DivMaker(
##     ButtonMaker()))






#' ButtonMaker
#'
#' This function crafts the actual button per se, including the specific
#' aesthetic traits for each button
#'
#' @param Color Optional numeric 1-6, corresponding to the colors specified in the bootstrap css classes:
#' \"success\",  \"warning\", \"danger\", \"info\", \"primary\", \"default\"
#' @param Size Optional numeric 1-4, corresponding to the sizes specified in the bootstrap css classes:
#' \"xs\",\"sm\",\"md\",\"lg\")
#' @param Value The numeric value you want to highlight (the main enchilada)
#' @param Subtitle Optional subtext that should appear under the value
#' @param Link Optional hyperlink that should be followed on click
#' @param Icon Optional glyphicon that should be displayed from http://getbootstrap.com/components/
#' @param Units Optional units that should be displayed after Value
#' @param Target Optional target that the value should be compared against. Use with ThresholdHigh and THresholdLow
#' @param ThresholdHigh Optional border between \"green\" and \"orange\". Use w/ Target and ThresholdLow. This value represents the RATIO
#' of the VALUE to the TARGET that, if above the ThresholdHigh will show as green, and if not, as orange
#' @param ThresholdLow Optional border between \"orange\" and \"red\". Use w/ Target and ThresholdLow. This value represents the RATIO
#' of the VALUE to the TARGET that, if above the ThresholdHigh will show as orange, and if not, as red
#' @param Hover Optional tooltip, or text that will show up when a user rests their mouse over the button.
#' @param alpha Optional transparency coefficient for the icon, a decimal from 0-1.
#' @param Former Optional value to compare against the current value. Will show up as a little arrow pointig the percent change
#'
#' @return Returns a character string containing HTML code to show the button, assuming the appropriate CSS elements will be available downstream
#' @examples
#' # ADD EXAMPLES HERE
#' # Button1 <- ButtonMaker(Color = 2,Value = 3.57,Subtitle = "B")
#' # Button2 <- ButtonMaker(Color = 3,Value = 13.7,Subtitle = "Nutritional value")
#' # Button3 <- ButtonMaker(Color = 4,Value = 1,Subtitle = "Yumminess factor")
#' # Button1;Button2;Button3
#' @export

ButtonMaker <- function(Color=1,Size=4,Value,Subtitle="",Link="",Icon="", Units="",
                        Target=0,ThresholdHigh=0,ThresholdLow=0, Hover="", alpha=0.5,
                        Former=Value){
  .Deprecated(solo_box, package="tileMaker", "These functions are provided for compatibility with older versions of R only, and may be
              defunct as soon as of the next release.",
              old = as.character(sys.call(sys.parent()))[1L])
  ## colors
  colorList = c("success",  "warning", "danger", "info", "primary", "default")

  ## sizes:
  SizeList = c("xs","sm","md","lg")

  paste(paste('<',
              if(Link !=""){paste('a href="',Link,'" role="button" ',sep='')} else{'button'},
              ' type="button" class="btn ',sep=''),
        if(Target ==0){
          paste('btn-',colorList[Color],sep='')
        } else {
          Perc <- Value/Target *100
          if(Perc > ThresholdHigh){
            'btn-success'
          } else if(Perc< ThresholdLow){
            'btn-danger'
          } else {
            'btn-warning'
          }
        },
        paste(' btn-', SizeList[Size],
              '"',

              if(Hover !=""){paste(' title="',Hover,'" ')},
              '><h1>',sep=''),

        if(Icon !=""){paste(' <span class="',Icon,'" aria-hidden="true" style="opacity:',alpha,'"></span> ',sep='')},
        paste(Value,Units,sep=''),

        if(Former>Value){
          paste('<sup style= "font-size: 12px;color:#EEEEEE;vertical-align: top;">&#9660;',round((Former-Value)/Former*100,1),'%</sup>',sep='')
        } else if (Former<Value){
          paste('<sup style= "font-size: 12px;color:#EEEEEE;vertical-align: top;">&#9650;',round((Value-Former)/Former*100,1),'%</sup>',sep='')
        },
        '</h1>',
        Subtitle,
        if(Link !=""){'</a>'} else{'</button>'},
        sep="")
}

#' DivMaker
#'
#' This function takes strings created with the function `ButtonMaker` and makes an HTML div suitable for inclusion in other HTML code,
#' or for inclusion within the function of this package `TileMaker`.
#'
#' @param Title The title for this row of buttons
#' @param Buttons The Buttons that you want inserted into this row. If you have more than one button, use paste(Button1,Button2)
#'
#' @return Returns an HTML string containing \"div\" elements. Beware of using these in Shiny... a it might break the container.
#' @examples
#' # ADD EXAMPLES HERE
#' Button1 <- ButtonMaker(Color = 2,Value = 3.57,Subtitle = "B")
#' Button2 <- ButtonMaker(Color = 3,Value = 13.7,Subtitle = "Nutritional value")
#' DivMaker(Title = "Quantativity factors",Buttons = paste(Button1,Button2))
#' @export

DivMaker <- function(Title="",Buttons){
  .Deprecated(div_maker, package="tileMaker", "These functions are provided for compatibility with older versions of R only, and may be
              defunct as soon as of the next release.",
              old = as.character(sys.call(sys.parent()))[1L])
  paste('<div class="container"><h2>',
        Title,
        '</h2>',
        Buttons,
        '</div>',sep="")
}


#' TileMaker
#'
#' Function 3 of 3, the last step. This function grabs the Divs created by `DivMaker` and
#' combines them into an html file.
#'
#' @param MainTitle Optional title for the whole set of titles
#' @param Divs The Divs that you want inserted into the tile. If you have more than one div,
#' use `paste(Div1,Div2)`
#' @param FileName The filename the tile should spit out as, including the extension (which
#' should always be html)
#' @param ShowDate Optional boolean controlling whether the date should be included or suppressed
#' @param localCSS Optional boolean to specify whether the bootstrap css file should be served from
#' the internet, or if you have saved a local version. If you have saved a local version, make sure
#' to download the "fonts" folder too, otherwise glyphicons won't work.
#'
#' @return Use this function to output an html file containing all the divs.If you would like just
#' HTML code (suitable for inserting in a dashboard or another document, you can use the Divs.
#'
#' @references Uses Twitter's awesome bootstrap V3
#'
#' @examples
#' # ADD EXAMPLES HERE
#' Button1 <- ButtonMaker(Color = 2,Value = 3.57,Subtitle = "B")
#' Button2 <- ButtonMaker(Color = 3,Value = 13.7,Subtitle = "Nutritional value")
#' Div1 <- DivMaker(Title = "Quantativity factors",Buttons = paste(Button1,Button2))
#' Div2 <- DivMaker(Title = "Inverse proportions",Buttons = paste(Button2,Button1))
#' TileMaker(MainTitle = "Hello",Divs = paste(Div1,Div2),FileName = "example.html")
#' TileMaker(MainTitle = "Hello",Divs = paste(Div1,Div2))
#' browseURL("example.html")
#' @export

TileMaker <- function(MainTitle="",Divs,FileName="x",ShowDate=FALSE,localCSS=FALSE){
  .Deprecated(file_maker, package="tileMaker", "These functions are provided for compatibility with older versions of R only, and may be
              defunct as soon as of the next release.",
              old = as.character(sys.call(sys.parent()))[1L])
  paste('<!DOCTYPE html><html lang="en"><head>
      <meta name="viewport" content="width=device-width, initial-scale=1">',
      if(localCSS==TRUE){'<link rel="stylesheet" href="bootstrap.min.css">'
      } else {'<link rel="stylesheet" href="http://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css">'},
      '</head><body><h1>',
      MainTitle,
      '</h1>',
      if(ShowDate){paste('<h2>Report Date: ',Sys.Date(),'</h2>',sep="")},
      Divs,
      '</body></html>',
      sep="") -> somethin

      ## Output file
      if (FileName !="x") {
        # sink(FileName)
        # somethin
        # sink()
        cat(somethin,file=FileName)
      } else {
          somethin
        }
}


R/tile_maker.R
# library(htmltools)
#' Tags
#' @importFrom htmltools tags browsable
#' @name tags
#' @export
#' @rdname TileMaker-exports
NULL

#' Auxiliary function to generate icons
#' @param x Icon name. See http://getbootstrap.com/components/
#' @param chevron binary to denote whether there is a former value to compare against or not.
#' @export
ico <- function(x,chevron=FALSE) {

  if(is.null(x)){
    NULL
    } else if(chevron==FALSE){
      tags$i(class = "glyphicon", class = paste0("glyphicon-", x))
    } else if(chevron==TRUE) {
      tags$i(class = "glyphicon",
             class = paste0("glyphicon-", x),
             style="font-size: 10px; vertical-align: top;")
    }
}

#' solo_box
#'
#' This function crafts the actual tile per se, including the specific
#' aesthetic traits for each tile. This is the simple version where you explicitly state the color.
#'
#' @param value The numeric value you want to highlight (the main enchilada)
#' @param former The numeric old value to use for comparison to 'value'
#' @param subtitle Optional subtext that should appear under the value
#' @param size Optional size specified in the bootstrap css classes:
#' \"xs\",\"sm\",\"md\",\"lg\")
#' @param icon Optional glyphicon that should be displayed from http://getbootstrap.com/components/
#' @param type Optional bootstrap css element that governs the color. https://v4-alpha.getbootstrap.com/utilities/colors/
#' Choose from: "Muted", "Primary", "Success", "Info", "Warning", "Danger"
#' @param link Optional hyperlink that should be followed on click
#' @param units Optional units that should be displayed after Value
#' @param hover Optional tooltip, or text that will show up when a user rests their mouse over the tile.
#' @importFrom htmltools HTML
#' @examples
#' tile1 <- solo_box(type="warning",value = 3.57,subtitle = "B")
#' tile2 <- solo_box(type="danger",value = 13.7,subtitle = "Nutritional value")
#' tile3 <- solo_box(type="success",value = 1,subtitle = "Yumminess factor")
#' tile4 <- solo_box(value = 3.57, former=3,subtitle = "Times apple eaten", icon = "apple")
#' file_maker(tile1,tile2,tile3,tile4)
#' file_maker(div_maker(tile1,tile2,tile3),div_maker(tile4))
#' @export solo_box
solo_box <- function(value = NULL, subtitle = NULL, former=NULL,size = "md", icon = NULL,
                    type = "info", link = NULL, units = NULL, hover = NULL) {

  tags$a(
    href = link,
    title = hover,
    # type = "button",
    type = type,
    role = "button",
    # classes: size, color
    class = "btn", class = paste0("btn-", size), class = paste0("btn-", type),
    tags$h1(ico(icon), value, units,
            if(!is.null(former)){
              if(former>value){
                tags$sup(style= "font-size: 12px;color:#EEEEEE;vertical-align: top;",
                         ico('chevron-down',chevron = T),paste(round((former-value)/former*100,1),'%',sep=''))
              } else {
                tags$sup(style= "font-size: 12px;color:#EEEEEE;vertical-align: top;",
                         ico('chevron-up',chevron = T),paste(round((former-value)/former*100,1),'%',sep=''))
              }
            }),
    HTML(subtitle)
  )
}


#' solo_gradient_box
#'
#' This function crafts a tile like solo_box, but this one changes color according to value
#'
#' @param value The numeric value you want to highlight (the main enchilada)
#' @param subtitle Optional subtext that should appear under the value
#' @param former The last value that should be used for comparison purposes
#' @param size Optional size specified in the bootstrap css classes:
#' \"xs\",\"sm\",\"md\",\"lg\")
#' @param icon Optional glyphicon that should be displayed from http://getbootstrap.com/components/
#' @param target Optional target that the value should be compared against. Use with ThresholdHigh and THresholdLow
#' @param thresholdHigh Optional edge between \"green\" and \"orange\" from 0-100. Use w/ Target and ThresholdLow. This value represents the RATIO
#' of the VALUE to the TARGET that, if above the ThresholdHigh will show as green, and if not, as orange
#' @param thresholdLow Optional border between \"orange\" and \"red\" from 0-100. Use w/ Target and ThresholdLow. This value represents the RATIO
#' of the VALUE to the TARGET that, if above the ThresholdHigh will show as orange, and if not, as red
#' @param link Optional hyperlink that should be followed on click
#' @param units Optional units that should be displayed after Value
#' @param hover Optional tooltip, or text that will show up when a user rests their mouse over the tile.
#' @importFrom htmltools HTML
#' @examples
#' # ADD EXAMPLES HERE
#' solo_gradient_box(value = 40)
#' solo_gradient_box(value = 40,target = 50,thresholdHigh = 80, thresholdLow=60)
#'
#' @export solo_gradient_box
solo_gradient_box <- function(value = NULL, subtitle = NULL, former=NULL, size = "md", icon = NULL,
                    target=100, thresholdHigh=90, thresholdLow=50,
                    link = NULL, units = NULL, hover = NULL) {

  Perc <- value/target *100
  if(Perc >= thresholdHigh){
    finalType='btn-success'
  } else if(Perc< thresholdLow){
    finalType='btn-danger'
  } else {
    finalType='btn-warning'
  }

  tags$a(
    href = link,
    title = hover,
    # type = "button",
    type = finalType,
    role = "button",
    # classes: size, color
    class = "btn", class = paste0("btn-", size), class = finalType,
    tags$h1(ico(icon), value, units,
            if(!is.null(former)){
              if(former>value){
                tags$sup(style= "font-size: 12px;color:#EEEEEE;vertical-align: top;",
                         ico('chevron-down',chevron = T),paste(round((former-value)/former*100,1),'%',sep=''))
              } else {
                tags$sup(style= "font-size: 12px;color:#EEEEEE;vertical-align: top;",
                         ico('chevron-up',chevron = T),paste(round((former-value)/former*100,1),'%',sep=''))
              }
            }),
    HTML(subtitle)
  )

}

#' Div maker
#'
#' This function takes strings created with the function `ButtonMaker` and makes an HTML div suitable for inclusion in other HTML code,
#' or for inclusion within the function of this package `TileMaker`.
#'
#' @param title Title.
#' @param ... \code{solo_box OR solo_gradient_box} elements.
#' @examples
#' div_maker(title = "Quantativity factors",
#'           solo_gradient_box(value = 70),
#'           solo_box(value=34))
#' @export div_maker
div_maker <- function(title = NULL, ...) {

  tags$div(
    class = "container",
    tags$h2(title),
    ...
  )

}


#' file maker
#'
#' Function 3 of 3, the last step. This function grabs the Divs created by `DivMaker` and
#' combines them into an html file.
#'
#' @param title Title.
#' @param ... \code{div_maker} elements.
#' @param css A string indicating css url
#' @param file Optional filename if you desire to save the file. Should end with ".html"
#' @importFrom htmltools browsable save_html
#' @export file_maker
file_maker <- function(title = NULL, ..., css = "https://bootswatch.com/flatly/bootstrap.css",
                       file=NULL) {

  tl <- tags$html(
    tags$head(
      tags$link(rel = "stylesheet", href = css)
    ),
    tags$body(
      tags$h1(title),
      ...
    )
  )

  if (is.null(file)) {
    browsable(tl)
  } else {
    save_html(tl,file = file)
  }
}


#' tileMatrix
#'
#' Create a matrix of buttons suitable for quick comparisons
#'
#' @param values Vector containing values for each tile
#' @param former vector containing former values (to show change from last)
#' @param subtitles Vector containing titles for each tile
#' @param tar Target value (What's the highest value to compare against). Defaults to 100
#' @param thre.H The limit between "high" and "medium" values IN PERCENT. Defaults to 90
#' @param thre.L The limit between "medium" and "low" values IN PERCENT. Defaults to 50
#' @param cols Number of columns that the matrix should tile around. Defaults to 4
#' @param mainTitle The title the matrix should have.
#' @param roundVal Number of decimals that Value will be rounded to. Defaults to 1
#' @importFrom htmltools HTML
#' @return Returns a list object containing the matrix of buttons
#' @examples
#' file_maker(tile_matrix(values=c(3,4,5),subtitles=c("Bob","Pedro","Ana")))
#' @export tile_matrix
tile_matrix <- function(values,subtitles,former=NULL,tar=100,thre.H=90,thre.L=50,cols=4,
                       mainTitle=NULL,roundVal=1){

  ## Errors
  if(class(values) != "numeric" & class(values) != "integer") stop("values should be numeric")
  if(class(subtitles) == "factor") subtitles <- as.character(subtitles)
  if(class(subtitles) != "character") stop("subtitles should be a character vector")
  if(length(values) != length(subtitles)) stop("values and subtitles vectors should be the same length, but they are not.")
  if(!is.null(former) & length(values) != length(former)) stop("'values' and 'former' vectors should be the same length, but they are not.")


  ## Clean inputs
  values <- round(values,roundVal)

  ## Make df and start adding extra stuffs
  if (is.null(former)){
    df <- data.frame(subtitles,values,stringsAsFactors = F)
  } else{
    df <- data.frame(subtitles,values,former,stringsAsFactors = F)
  }

  df$id <- 1:nrow(df)
  df$butts <- list("")

  # df$stuff <- str_trunc(df$stuff,min(str_length(df$stuff)),side="right")

  ## protect against NAs
  if(any(is.na(df$value))){
    df$value[is.na(df$value)] <-0.001
    warning("Converted NAs in values to 0.001")
  }
  if(!is.null(former) ) {
    df$former[is.na(df$former)] <-0.001
    warning("Converted NAs in former to 0.001")
  }

  for(i in 1:nrow(df)){
      if(!is.null(former)){
        df$butts[[i]] <- solo_gradient_box(value = df$value[i],subtitle = df$subtitles[i],
                                         size = 2,target=tar,thresholdHigh = thre.H,
                                   thresholdLow = thre.L,former=df$former[i])
      } else {
        df$butts[[i]] <- solo_gradient_box(value = df$value[i],subtitle = df$subtitles[i],
                                           size = 2,target=tar,thresholdHigh = thre.H,
                                           thresholdLow = thre.L)
      }
  }

  Sausage = df$butts

  # ## Break the button sausage every cols
  splitter <- function(Sausage,cols){
    Outputter <- list("")
    for (i in 1:ceiling(length(Sausage)/cols)){
      Outputter[[i]] <- div_maker(Sausage[((i-1)*cols+1):(cols*i)])
    }
    Outputter
  }

  tags$a(
    tags$h1(HTML(mainTitle)),
    splitter(Sausage,cols)
  )
}

#' @title multi_box
#' @description Create a tile that contains more than one value, icon and units
#' @param MB_values vector of values to display, Default: NULL
#' @param mainTitle Top title, Default: NULL
#' @param size Optional size specified in the bootstrap css classes:
#' \"xs\",\"sm\",\"md\",\"lg\")
#' @param MB_icons vector of Icons to display, Default: NULL
#' @param type Optional bootstrap css element that governs the color. https://v4-alpha.getbootstrap.com/utilities/colors/
#' Choose from: "Muted", "Primary", "Success", "Info", "Warning", "Danger", Default: 'info'
#' @param link Optional hyperlink to redirect to after a user click, Default: NULL
#' @param MB_units vector of Units that explain the values, Default: NULL
#' @param hover Optional tooltip, or text that will show up when a user rests their
#' mouse over the tile, Default: NULL
#' @importFrom purrr pmap
#' @importFrom htmltools HTML
#' @return OUTPUT_DESCRIPTION
#' @details DETAILS
#' @examples
#' file_maker(multi_box(MB_values=c(3,45),mainTitle = "Important <br>button",
#' MB_icons=c("apple","calendar"), type="warning", MB_units=c("times","reports")))
#' \dontrun{
#' if(interactive()){
#'  #EXAMPLE1
#'  }
#' }
#' @rdname multi_box
#' @export

multi_box <- function(MB_values = NULL, mainTitle = NULL ,size = "md", MB_icons = NULL,
                     type = "info", link = NULL, MB_units = NULL, hover = NULL) {
  ## Define function that can be pmapped
  gutsMaker <- function(MB_values,MB_units,MB_icons){
    tags$h3(ico(MB_icons), MB_values, MB_units)
  }

  ## Now build button
  tags$a(
    href = link,
    title = hover,
    # type = "button",
    type = type,
    role = "button",
    # classes: size, color
    class = "btn", class = paste0("btn-", size), class = paste0("btn-", type),
    tags$h1(HTML(mainTitle)),
    pmap(list(MB_values,MB_units,MB_icons),gutsMaker)
  )
}


# library(dplyr)
# library(htmltools)
# a <- tile_matrix(values = iris$Sepal.Length,titles = iris$Species,tar = 5,thre.H = 4,thre.L = 3)

# ico(NULL)
# ico("apple")
#
# b1 <- solo_box(value = 3.57, subtitle = "Times apple eaten", icon = "apple")
# b2 <- solo_box(value = 13.7, subtitle = "Nutritional value", size = "lg")
# b3 <- solo_box(value = 1, subtitle = "Yumminess factor", type = "danger")
# b4 <- solo_box(value = 5, subtitle = "Inconsistencies", hover = "This is the description")
#
# d1 <- div_maker(title = "Quantativity factors", b1, b2)
# d2 <- div_maker(title = "Implementation procedures", b3, b4)
#
# file_maker(title = "Hello", b1, b2)
#
# map <- leaflet::addTiles(leaflet::leaflet())
# map
#
# file_maker(title = "Hello", d1, map, d2)
#
#
# ## Shinydashboard Test --------------
# ## app.R ##
# library(shinydashboard)
# library(shiny)
#
# ui <- dashboardPage(
#   dashboardHeader(title = "Basic dashboard"),
#   dashboardSidebar(),
#   dashboardBody(
#     # Boxes need to be put in a row (or column)
#     fluidRow(
#       box(plotOutput("plot1", height = 250)),
#
#       box(
#         title = "Controls",
#         d1,
#         d2,
#         sliderInput("slider", "Number of observations:", 1, 100, 50)
#       ),
#       box(d1),
#       box(d2),
#       box(d1,d2)
#     )
#   )
# )
#
# server <- function(input, output) {
#   set.seed(122)
#   histdata <- rnorm(500)
#
#   output$plot1 <- renderPlot({
#     data <- histdata[seq_len(input$slider)]
#     hist(data)
#   })
# }
#
# shinyApp(ui, server)
#
# ## shinyTest ---------
# server <- function(input, output) {
#   output$distPlot <- renderPlot({
#     hist(rnorm(input$obs), col = 'darkgray', border = 'white')
#   })
# }
#
# ui <- fluidPage(
#   sidebarLayout(
#     sidebarPanel(
#       sliderInput("obs", "Number of observations:", min = 10, max = 500, value = 100)
#     ),
#     mainPanel(d1,d2,plotOutput("distPlot"),
#               fluidRow(d1,d2))
#   )
# )
#
# shinyApp(ui = ui, server = server)



